new topic for Workload Protection Hosts and Containers (#31508)
DataDog/documentation: content/en/security/workload_protection/inventory/hosts_and_containers.md

---
title: Hosts and Containers
disable_toc: false
---

> The [Hosts and Containers][1] inventory in Datadog Workload Protection provides a unified visibility layer for host-level agent deployment, configuration health, and security feature status. 

Reviewer: Nit for directness

```suggestion
The [Hosts and Containers][1] inventory in Datadog Workload Protection provides a unified view of host-level agent deployment, configuration health, and security feature status. 
```

**Hosts and Containers** enables DevSecOps teams to verify whether the following protections are properly deployed and running across environments:

- [Workload Protection][3]
- [CSM Misconfigurations][2]
- [Container Vulnerability Scanning][4]
- [Host Vulnerability Scanning][4]
- It also helps identify which hosts and containers are running old versions of the Agent.

> **Hosts and Containers** also provides remediation guidance for missing protections.

Reviewer: The last bullet item in the bulleted list isn't parallel with the others, and doesn't naturally follow from the lead-in sentence. The last sentence here on line 16 also feels like a bit of an orphan. I'd recommend either:

Option 1: fold last bulleted item into the sentence on line 16:
```
**Hosts and Containers** enables DevSecOps teams to verify whether the following protections are properly deployed and running across environments:

- [Workload Protection][3]
- [CSM Misconfigurations][2]
- [Container Vulnerability Scanning][4]
- [Host Vulnerability Scanning][4]

In addition, **Hosts and Containers** helps you identify which hosts and containers are running old versions of the Agent, and provides remediation guidance for missing protections.
```

Option 2: turn the whole set of capabilities into a bulleted list:
```
**Hosts and Containers** enables DevSecOps teams to:

- Verify that protections are properly deployed and running across environments, including:
  - [Workload Protection][3]
  - [CSM Misconfigurations][2]
  - [Container Vulnerability Scanning][4]
  - [Host Vulnerability Scanning][4]
- Identify which hosts and containers are running old versions of the Agent
- Access remediation guidance for missing protections
```

Author: Thanks. I just meant it as a sentence and not part of the the list or subordinate to the lead in. It's just stating that remediation is also offered.

That said, your second option is good. I'll use that. thanks!

## Use cases

> Here are some common DevSecOps use cases for the [Hosts and Containers][1] inventory feature.

Reviewer: Don't love the "here are" but not a dealbreaker by any means! Defer to you.

```suggestion
The [Hosts and Containers][1] inventory supports several common DevSecOps use cases.
```

### Assess coverage status

> To identify hosts where runtime threat detection is not configured, do the following:

Reviewer: ```suggestion
To identify hosts where runtime threat detection is not configured:
```

1. In [Hosts and Containers][1], set the following facets to **false**:
   - **Workload Protection Enabled**
   - **Misconfigurations Enabled**
   - **Hosts VM Enabled**
   - **Containers VM Enabled**
   

>    The hosts and containers missing one or more of these features are shown with an orange icon. This list flags coverage gaps that expose the workload to undetected process, file, or network-level threats.

Reviewer: ```suggestion
   The hosts and containers missing one or more of these features are indicated by an orange icon. This view highlights coverage gaps that can expose workloads to undetected process-, file-, or network-level threats.
```

2. For remediation guidance, hover over the icon for a feature and click **Configure**.

<div class="alert alert-info">

> Filter by <b>Containers VM Enabled: true</b> to ensure scanning is applied to ephemeral container workloads too.

Reviewer: Flows a bit better this way imo but defer to you!

```suggestion
Filter by <b>Containers VM Enabled: true</b> to ensure scanning is also applied to ephemeral container workloads.
```

</div>

### Validate Agent health

> To validate Agent health, do the following:

Reviewer: ```suggestion
To validate Agent health:
```

1. In the **Agent Version** column, look for older versions identified with yellow labels.
   

>    Yellow labels identify that these versions might not support all security features. 

Reviewer: ```suggestion
   Yellow labels indicate versions that might not support all security features.
```

2. Click a version label (for example, 7.69.1), and select **Filter by agent_version:[number]**. 
   This isolates all hosts running that version. 

> 3. Outside of [Hosts and Containers][1], check each Agent host for upgrade readiness and [schedule upgrades accordingly][5].

Reviewer: "Outside of" is a little vague. I think cutting that out is okay.

```suggestion
3. Check each Agent host for upgrade readiness and [schedule upgrades accordingly][5].
```

Author: I wanted the "Outside of" to make it clear that it's a process outside of the ability of this page, and Datadog in fact.

> ### Detect misconfigurations

Author: they're on the page so I think we should, but I'll point them to the other doc.

> Hosts without CSM Misconfigurations enabled can't surface IAM, logging, or encryption misconfigurations. Misconfiguration checks are critical for CIS Benchmarks or NIST-aligned cloud security posture management.

Reviewer: ```suggestion
Hosts without CSM Misconfigurations enabled can't surface IAM, logging, or encryption misconfigurations. Misconfiguration checks are critical for CIS Benchmarks and NIST-aligned cloud security posture management.
```

> To see if posture checks are enabled for a host, do the following:

Reviewer: ```suggestion
To check whether posture checks are enabled for a host:
```

1. In [Hosts and Containers][1], set the **Misconfigurations Enabled** facet to **false**.

>    The hosts and containers without CSM Misconfigurations enabled are shown with an orange icon. 

Reviewer: ```suggestion
   The hosts and containers without CSM Misconfigurations enabled are indicated by an orange icon. 
```

2. For remediation guidance, hover over the **CSM Misconfigurations** icon and click **Configure**.

### Cluster-level tracking

> The **Cluster Name** column links hosts to logical infrastructure boundaries like Kubernetes clusters. You can filter on a cluster by clicking its name and selecting **Filter by cluster_name:[name]**.

Reviewer: ```suggestion
The **Cluster Name** column links hosts to logical infrastructure boundaries like Kubernetes clusters. Filter on a cluster by clicking its name and selecting **Filter by cluster_name:[name]**.
```

> Filtering on a cluster confirms whether protections are applied uniformly. This helps prevent misalignment between environments or inconsistent deployments by region.

Reviewer: ```suggestion
Filtering on a cluster confirms whether protections are applied uniformly. This ensures protections are applied consistently across environments and regions.
```

### Prioritize response

> [Hosts and Containers][1] can be used when investigating alerts or during incident triage. Use the feature icon panel to spot protection gaps. 

Reviewer: Tweak to make this active voice

```suggestion
[Hosts and Containers][1] supports investigations and incident triage. Use the feature icon panel to spot protection gaps. 
```

Typically, hosts missing critical features like **Workload Protection** or **Host/Container Vulnerability Scanning** are triaged first. 

> Hover over a feature icon and click **Configure** to see remediation steps for all feature gaps. This let's you perform triage without context switching.

Reviewer: If you want to keep "This let's you", just remove the apostrophe -> "This lets you"

```suggestion
Hover over a feature icon and click **Configure** to see remediation steps for all feature gaps. This enables triage without context switching.
```

### Compliance evidence

> [Hosts and Containers][1] serves as a live audit view for workload security posture. Filters such as **Workload Protection Enabled** and **Agent Version** demonstrate control coverage to auditors for frameworks like SOC 2, PCI DSS, or FedRAMP.

Reviewer: A few minor tweaks. Also I think "auditors" is implied by the frameworks so can be removed.

```suggestion
[Hosts and Containers][1] provides a live audit view for workload security posture. Filters such as **Workload Protection Enabled** and **Agent Version** demonstrate control coverage for frameworks such as SOC 2, PCI DSS, or FedRAMP.
```
